File: 03-march/30-check-if-strings-can-be-made-equal-with-operations-ii/solution.py
class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        counts = [0] * 52

        for i, (char1, char2) in enumerate(zip(s1, s2)):
            offset = 26 if i & 1 else 0
            counts[offset + ord(char1) - ord("a")] += 1
            counts[offset + ord(char2) - ord("a")] -= 1

        return all(count == 0 for count in counts)

File: 03-march/30-check-if-strings-can-be-made-equal-with-operations-ii/solution.java
class Solution {
    public boolean checkStrings(String s1, String s2) {
        int[] counts = new int[52];

        for (int i = 0; i < s1.length(); i++) {
            int offset = (i & 1) == 0 ? 0 : 26;
            counts[offset + s1.charAt(i) - 'a']++;
            counts[offset + s2.charAt(i) - 'a']--;
        }

        for (int count : counts) {
            if (count != 0) {
                return false;
            }
        }

        return true;
    }
}
